### PR TITLE
DOP-4491: Ensure that cache data is invalidated when images change

### PR DESCRIPTION
**Pull request by i80and**
This removes the expensive_operation_cache, which is very old. The purpose is to avoid re-hashing image assets in the vscode use-case, but I don't think the benefit is worth having this weird interface around.

We should revisit in a post free-threading Python world when we can actually give workers access to shared state. That would also let us get rid of the entire pending task concept.

### Ticket

DOP-4491

### README updates

- - [ ] This PR introduces changes that should be reflected in the README.md and/or HACKING.md, and I have made those updates.
- - [x] This PR does not introduce changes that should be reflected in the README.md and/or HACKING.md

### CODE REVIEW

File: snooty/parser.py [hunk 1]
```diff
@@ -144,32 +143,33 @@ def verify(self) -> None:
...
+    ) -> None:
         super().__init__(node)
         self.node: n.Directive = node
         self.asset = asset
+        self.dependencies = dependencies
 
     def __call__(
         self, diagnostics: List[Diagnostic], project: ProjectInterface
     ) -> None:
         """Compute this figure's checksum and store it in our node."""
```
Comment (seungpark) [inline snooty/parser.py:160]: minor. can we update the comments here?

File: snooty/test_project.py [hunk 2]
```diff
@@ -417,6 +391,65 @@ def test_cache() -> None:
                     )
 
 
+def test_image_invalidation() -> None:
```
Comment (seungpark) [inline snooty/test_project.py:394]: awesome! TIL programmatically testing the cache. previous cache is not used for changed image 👍

**Review by seungpark — APPROVED** [review]
thank you for the explanation, much appreciated! will have to fix this [minor svg file](https://github.com/10gen/docs-k8s-operator/blob/master/source/images/kubernetes-operator.bakedsvg.svg?short_path=dfbf22d) for ([this error](https://mongodb.slack.com/archives/C018UTGP23T/p1711377002896719) is actually correct in that it can't compute intrinsic size, but now previous cached image will be updated 🙏  )

File: snooty/parser.py (hunk 1 above)
Comment (i80and) [reply snooty/parser.py:160]: oop good eye!

**Pull request merged**